src/http/UnsecureWebSocketsProtocol.ts

```diff
@@ -47,7 +47,7 @@ class WebSocketListener extends EventEmitter {
     // Store the HTTP host and protocol
     const forwarded = parseForwarded(headers);
     this.host = forwarded.host ?? headers.host ?? 'localhost';
-    this.protocol = forwarded.proto === 'https' || (socket as any).secure ? 'https:' : 'http:';
+    this.protocol = forwarded.proto === 'https' || (socket as any).encrypted ? 'https:' : 'http:';
```

Comment:
```suggestion
    this.protocol = forwarded.proto === 'https' || (socket as TLSSocket).encrypted ? 'https:' : 'http:';
```
Having correct typing there would make sure this stays correct.